CoreFoundation/NumberDate.subproj/CFDate.c

```diff
@@ -171,10 +171,11 @@ CF_PRIVATE void __CFDateInitialize(void) {
     __CFTSRRate = (1.0E9 / (double)info.numer) * (double)info.denom;
     __CF1_TSRRate = 1.0 / __CFTSRRate;
 #elif TARGET_OS_WIN32
-    // We are using QueryUnbiasedInterruptTimePrecise as time source.
-    // It returns result in system time units of 100 nanoseconds. 
-    // To get seconds we need to divide the value by 1e7 (10000000).
-    __CFTSRRate = 1.0e7;
+    LARGE_INTEGER freq;
+    if (!QueryPerformanceFrequency(&freq)) {
+        HALT;
+    }
+    __CFTSRRate = (double)freq.QuadPart;
```

Comment:
This was modified in #2907. Long story short, CFRunLoop is sensitive to the way of using system timers. We have to keep it consistent with Dispatch, or CFRunLoop would rarely and randomly hang.

Reply:
Understood; will revert.